corehq/apps/userreports/reports/data_source.py

```diff
@@ -163,8 +164,16 @@ def has_total_row(self):
         return any(column_config.calculate_total for column_config in self.column_configs)
 
     def get_total_records(self):
-        # TODO - actually use sqlagg to get a count of rows
-        return len(self.get_data())
+        qc = self.query_context()
+        for c in self.columns:
+            qc.append_column(c.view)
```

Comment:
the reason we need to add all the columns is so the filters and group-by get applied, yeah? might be worth a comment (and noting that in the future we could remove the columns that are being aggregated but not part of filters or group bys)